zulipterminal/ui_tools/views.py

```diff
@@ -807,37 +858,15 @@ def __init__(self, controller: Any,
 
 
 class HelpView(PopUpView):
-    def __init__(self, controller: Any) -> None:
-        widths = [(len(binding['help_text']) + 4,
-                   len(", ".join(binding['keys'])))
-                  for binding in KEY_BINDINGS.values()]
-        max_widths = [max(width) for width in zip(*widths)]
-        self.width = sum(max_widths)
-
-        help_menu_content = []  # type: List[urwid.AttrWrap]
+    def __init__(self, controller: Any, title_len: int) -> None:
+        help_menu_content = []
         for category in HELP_CATEGORIES:
-            if len(help_menu_content) > 0:  # Separate categories by newline
-                help_menu_content.append(urwid.Text(''))
-
-            help_menu_content.append(urwid.Text(('popup_category',
-                                                 HELP_CATEGORIES[category])))
-
-            keys_in_category = (binding for binding in KEY_BINDINGS.values()
-                                if binding['key_category'] == category)
-            for help_item_number, binding in enumerate(keys_in_category):
-                help_menu_content.append(
-                    urwid.AttrWrap(
-                        urwid.Columns([
-                            urwid.Text(binding['help_text']),
-                            (max_widths[1],
-                                urwid.Text(", ".join(binding['keys'])))
-                            ], dividechars=2),
-                        None if help_item_number % 2 else 'popup_contrast'
-                    )
-                )
-
-        self.height = len(help_menu_content)
-
+            help_menu_content.append((HELP_CATEGORIES[category], [
+                (binding['help_text'], ', '.join(binding['keys']))
+                for binding in [binding for binding in KEY_BINDINGS.values()
+                                if binding['key_category'] == category]
+            ]))
```

Comment:
You've compacted the generation of this data-structure, and while it's great that the data is more localized than it was when mixed with the UI elements, it's not quite as clear as it was before - it has a lot of nested elements.